### PR TITLE
Add OpenTracing support for Kafka

### PR DESCRIPTION
**Pull request by loicmathieu**
This PR adds two distinct features:
- Register for reflection all Kafka consumer/producer interceptor so they can be used in native
- Integrate Kafka OpenTracing contributor interceptors inside Quarkus (doc, bom, register for reflection)

See the Zulip thread for reference: https://quarkusio.zulipchat.com/#narrow/stream/187038-dev/topic/OpenTracing.20Kafka.20intereptors

### CODE REVIEW

File: docs/src/main/asciidoc/opentracing.adoc [hunk 1]
```diff
@@ -196,6 +196,38 @@ quarkus.datasource.jdbc.driver=io.opentracing.contrib.jdbc.TracingDriver
 quarkus.hibernate-orm.dialect=org.hibernate.dialect.PostgreSQLDialect
 ----
 
+
+=== Kafka
+
+The https://github.com/opentracing-contrib/java-kafka-client[Kakfa instrumentation] will add a span for each message sent to or received from a Kafka topic, to enable it, add the following dependency to your pom.xml:
```
Comment (Ladicek) [inline docs/src/main/asciidoc/opentracing.adoc:202]:
```suggestion
The https://github.com/opentracing-contrib/java-kafka-client[Kafka instrumentation] will add a span for each message sent to or received from a Kafka topic. To enable it, add the following dependency to your pom.xml:
```

File: docs/src/main/asciidoc/opentracing.adoc [hunk 2]
```diff
@@ -196,6 +196,38 @@ quarkus.datasource.jdbc.driver=io.opentracing.contrib.jdbc.TracingDriver
...
+<dependency>
+    <groupId>io.opentracing.contrib</groupId>
+    <artifactId>opentracing-kafka-client</artifactId>
+    <version>${opentracing-kafka.version}</version>
+</dependency>
+----
+
+It contains OpenTracing interceptors that must be registered on Kafka producers and consumers.
+
+If you followed the link:kafka[Kafka guide], the interceptors can be added on the `generated-price` and the `prices` channels as follows:
```
Comment (cescoffier) [inline docs/src/main/asciidoc/opentracing.adoc:215]: Maybe mentioned that interceptors are comma-separated (if I'm not mistaken)

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/opentracing.adoc (hunk 2 above)
Comment (loicmathieu) [reply docs/src/main/asciidoc/opentracing.adoc:215]: I added a note

**Comment by loicmathieu** [comment]
@Ladicek I applied your suggestions.

File: docs/src/main/asciidoc/opentracing.adoc (hunk 2 above)
Comment (cescoffier) [reply docs/src/main/asciidoc/opentracing.adoc:215]: Thanks!

**Review by cescoffier — APPROVED** [review]

**Pull request merged**